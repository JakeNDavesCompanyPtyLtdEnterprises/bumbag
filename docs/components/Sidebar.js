import React from 'react';
import { Box, Icon, Hidden, Menu as _Menu, Sidebar as PopoutSidebar, palette, space, styled, theme } from 'fannypack';
import Link from 'next/link';

import { useDocsContext } from './DocsContext';

const Spacer = styled(Box)`
  width: 250px;

  @media screen and (max-width: ${theme('fannypack.layout.desktopBreakpoint')}px) {
    width: 0px;
  }
`;
const Wrapper = styled(Box)`
  background-color: ${palette('white700')};
  height: 100vh;
  min-width: 250px;
  width: 250px;
  overflow-y: scroll;
  transform: translateX(0px);

  @media screen and (max-width: ${theme('fannypack.layout.desktopBreakpoint')}px) {
    transform: translateX(${props => (props.isMenuOpen ? '0px' : '-250px')});
  }
`;
const Menu = styled(_Menu)`
  && button {
    border-radius: 3px;
    margin-bottom: ${space(1)}rem;

    &:hover {
      background-color: unset;
      color: ${palette('primary')};
    }

    &:focus {
      background-color: unset;
    }
  }
`;

function Sidebar() {
  const { layout = {}, routes = [], route = {} } = useDocsContext();

  const MobileSidebar = React.useCallback(
    ({ children }) => (
      <PopoutSidebar isVisible={layout.isMenuOpen} onClickClose={layout.closeMenu}>
        {children}
      </PopoutSidebar>
    ),
    [layout]
  );
  const DesktopSidebar = React.useCallback(({ children }) => <Box fixed>{children}</Box>);
  const SidebarContainer = layout.isMobile ? MobileSidebar : DesktopSidebar;

  const menuItems = routes.reduce((currentMenuItems, route) => {
    let newMenuItems = currentMenuItems;
    if (route.menu) {
      if (!newMenuItems[route.menu]) {
        newMenuItems[route.menu] = [route];
      } else {
        newMenuItems[route.menu].push(route);
      }
    } else {
      newMenuItems[route.name] = route;
    }
    return newMenuItems;
  }, {});

  return (
    <React.Fragment>
      <Spacer />
      <SidebarContainer>
        <Wrapper isMenuOpen={layout.isMenuOpen}>
          <Menu margin="major-2">
            {Object.entries(menuItems).map(([name, menuItem], i) => (
              <React.Fragment key={i}>
                {Array.isArray(menuItem) ? (
                  <Hidden.Container defaultVisible={route.menu === name}>
                    {hidden => (
                      <React.Fragment>
                        <Menu.Item
                          display="flex"
                          alignItems="center"
                          justifyContent="space-between"
                          onClick={hidden.toggle}
                        >
                          {name}
                          <Icon icon={hidden.isVisible ? 'solid-chevron-up' : 'solid-chevron-down'} top="0" />
                        </Menu.Item>
                        {hidden.isVisible && (
                          <Box paddingLeft="major-2" width="100%">
                            {menuItem.map((item, i) => (
                              <Link key={i} href={item.path}>
                                <Menu.Item color={route.path === item.path ? 'primary' : undefined} fontWeight="400">
                                  {item.name}
                                </Menu.Item>
                              </Link>
                            ))}
                          </Box>
                        )}
                      </React.Fragment>
                    )}
                  </Hidden.Container>
                ) : (
                  <Link href={menuItem.path}>
                    <Menu.Item color={route.path === menuItem.path ? 'primary' : undefined}>{name}</Menu.Item>
                  </Link>
                )}
              </React.Fragment>
            ))}
          </Menu>
        </Wrapper>
      </SidebarContainer>
    </React.Fragment>
  );
}

export default Sidebar;
