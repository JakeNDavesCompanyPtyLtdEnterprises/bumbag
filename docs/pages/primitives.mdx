import { Box, Block, Inline, InlineBlock, InlineFlex, Grid, Flex } from 'fannypack';

# Primitives

Fannypack exports Reakit's primitives:

- `<Box>`
- `<Block>`
- `<Inline>`
- `<InlineBlock>`
- `<Flex>`
- `<InlineFlex>`
- `<Grid>`

These primitive components provide an abstraction on top of the `<div>` element to include reset styles and allows you to specify CSS attributes as props (see the below for more details).

See more about primitive components [here](https://reakit.io/components/box).

## Box

```jsx
import { Box } from 'fannypack'
```

All components in Fannypack are built on top of the `<Box>` primitive. Any component in Fannypack can use `Box`'s props. [Read more here](https://reakit.io/components/box).

By default it renders a `<div>` element with basic reset styles.

```.jsx
<Box>This is a box</Box>
```

You're able to apply any CSS attribute as a prop to box:

```.jsx
<Box backgroundColor="primary" color="white" padding="major-3">
  This is a box
</Box>
```

#### Color attributes

Any CSS color attribute (e.g. `color`, `border-color`, etc) can take a [color from the Fannypack palette](/palette#theming).

```.jsx
<Box backgroundColor="primaryTint" color="primary">
  This is a box
</Box>
```

#### Margin & padding attributes

Any CSS margin/padding attribute (e.g. `margin`, `padding-left`, etc) can take a [spacing from the Fannypack theme](/styling#layoutthemeconfig).

```.jsx
<Box backgroundColor="primaryTint" padding="major-3">
  This is a box
</Box>
```

#### Font attributes

The `fontWeight` CSS prop can contain of a [font weight in the Fannypack theme](/styling#fontweightsthemeconfig).

```.jsx
<Box fontWeight="semibold">This is a box</Box>
```

The `fontSize` CSS prop can contain of a [font size in the Fannypack theme](/styling#fontsizethemeconfig).

```.jsx
<Box fontSize="400">This is a box</Box>
```

## Block

```jsx
import { Block } from 'fannypack'
```

A `<Block>` is a `<Box>` with `display: block`. [Read more here](https://reakit.io/components/block).

```.jsx
<Block backgroundColor="primaryTint" padding="major-3">
  Block 1
</Block>
<Block backgroundColor="successTint" padding="major-3">
  Block 2
</Block>
<Block backgroundColor="dangerTint" padding="major-3">
  Block 3
</Block>
```

## Inline

```jsx
import { Inline } from 'fannypack'
```

An `<Inline>` is a `<Box>` with `display: inline`. [Read more here](https://reakit.io/components/inline).

```.jsx
<Inline>
  Inline 1
</Inline>
<Inline>
  Inline 2
</Inline>
<Inline>
  Inline 3
</Inline>
```

## InlineBlock

```jsx
import { InlineBlock } from 'fannypack'
```

An `<InlineBlock>` is a `<Box>` with `display: inline-block`. [Read more here](https://reakit.io/components/inlineblock).

```.jsx
<InlineBlock backgroundColor="primaryTint" padding="major-3">
  InlineBlock 1
</InlineBlock>
<InlineBlock backgroundColor="successTint" padding="major-3">
  InlineBlock 2
</InlineBlock>
<InlineBlock backgroundColor="dangerTint" padding="major-3">
  InlineBlock 3
</InlineBlock>
```

## Flex

```jsx
import { Flex } from 'fannypack'
```

A `<Flex>` is a `<Box>` with `display: flex`. [Read more here](https://reakit.io/components/flex).

```.jsx
<Flex>
  <Box backgroundColor="primaryTint" padding="major-3">
    Box 1
  </Box>
  <Box backgroundColor="successTint" padding="major-3">
    Box 2
  </Box>
  <Box backgroundColor="dangerTint" padding="major-3">
    Box 3
  </Box>
</Flex>
```

## InlineFlex

```jsx
import { InlineFlex } from 'fannypack'
```

An `<InlineFlex>` is a `<Box>` with `display: inline-flex`. [Read more here](https://reakit.io/components/inlineflex).

```.jsx
<InlineFlex justifyContent="space-evenly" width="100%">
  <Box backgroundColor="primaryTint" padding="major-3">
    Box 1
  </Box>
  <Box backgroundColor="successTint" padding="major-3">
    Box 2
  </Box>
  <Box backgroundColor="dangerTint" padding="major-3">
    Box 3
  </Box>
</InlineFlex>
```

## Grid

```jsx
import { Grid } from 'fannypack'
```

A `<Grid>` is a `<Box>` with `display: grid`. It also consists of a `<Grid.Item>` component to pair with the `<Grid>`. [Read more here](https://reakit.io/components/grid).

```.jsx
<Grid templateColumns="repeat(2, 1fr)" autoRows="auto" gap="3vw">
  <Grid.Item backgroundColor="primaryTint" padding="major-3">
    Grid Item 1
  </Grid.Item>
  <Grid.Item backgroundColor="successTint" padding="major-3">
    Grid Item 2
  </Grid.Item>
  <Grid.Item backgroundColor="dangerTint" padding="major-3">
    Grid Item 3
  </Grid.Item>
</Grid>
```
