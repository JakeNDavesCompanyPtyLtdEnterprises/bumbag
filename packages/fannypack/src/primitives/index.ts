export { default as Box } from 'reakit/Box';
export { default as Block } from 'reakit/Block';
export { default as Inline } from 'reakit/Inline';
export { default as InlineBlock } from 'reakit/InlineBlock';
export { default as Flex } from 'reakit/Flex';
export { default as InlineFlex } from 'reakit/InlineFlex';
export { default as Grid } from 'reakit/Grid';
