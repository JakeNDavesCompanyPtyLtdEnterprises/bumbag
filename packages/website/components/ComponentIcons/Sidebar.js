import React from 'react';

const SvgComponent = props => (
  <svg width="89" height="73" viewBox="0 0 89 73" fill="none" xmlns="http://www.w3.org/2000/svg">
    <rect y="1" width="72" height="72" fill="white" />
    <path
      d="M81.772 6.32L79.216 1.982H83.014L84.022 4.106L85.066 1.982H88.756L86.236 6.32L88.936 11H85.246L84.022 8.696L82.834 11H79.036L81.772 6.32Z"
      fill="white"
    />
    <rect x="10" y="14" width="52" height="5" fill="#3926A5" />
    <rect x="10" y="24" width="30" height="5" fill="#3926A5" />
    <rect x="10" y="34" width="41" height="5" fill="#3926A5" />
    <rect x="10" y="44" width="35" height="5" fill="#3926A5" />
    <rect x="10" y="54" width="46" height="5" fill="#3926A5" />
  </svg>
);

export default SvgComponent;
