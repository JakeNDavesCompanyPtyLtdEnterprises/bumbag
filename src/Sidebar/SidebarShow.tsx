import * as React from 'react';
import * as PropTypes from 'prop-types';
import { SidebarShowProps as ReakitSidebarShowProps } from 'reakit/ts';

import { SidebarShow as _SidebarShow } from './styled';

export type LocalSidebarShowProps = {
  children?: React.ReactNode | void;
  onClick?(): void;
  show(): void;
};
export type SidebarShowProps = LocalSidebarShowProps & ReakitSidebarShowProps;

export const SidebarShow: React.FunctionComponent<LocalSidebarShowProps> = ({ children, ...props }) => (
  <_SidebarShow {...props}>{children}</_SidebarShow>
);

export const sidebarShowPropTypes = {
  children: PropTypes.node,
  onClick: PropTypes.func,
  show: PropTypes.func.isRequired
};
SidebarShow.propTypes = sidebarShowPropTypes;

export const sidebarShowDefaultProps = {
  children: null,
  onClick: undefined
};
SidebarShow.defaultProps = sidebarShowDefaultProps;

// @ts-ignore
const C: React.FunctionComponent<SidebarShowProps> = SidebarShow;
export default C;
