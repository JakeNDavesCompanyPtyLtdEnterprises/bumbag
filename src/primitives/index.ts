// @ts-ignore
export { default as Box } from 'reakit/Box';
// @ts-ignore
export { default as Block } from 'reakit/Block';
// @ts-ignore
export { default as Inline } from 'reakit/Inline';
// @ts-ignore
export { default as InlineBlock } from 'reakit/InlineBlock';
// @ts-ignore
export { default as Flex } from 'reakit/Flex';
// @ts-ignore
export { default as InlineFlex } from 'reakit/InlineFlex';
// @ts-ignore
export { default as Grid } from 'reakit/Grid';
