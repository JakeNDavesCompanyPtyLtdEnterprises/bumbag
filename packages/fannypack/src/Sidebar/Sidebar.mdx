---
name: Sidebar
route: /utils/sidebar
menu: Utilities
---

import { Playground } from 'docz';
import PropsTable from '../_docs/components/PropsTable';
import { Box } from '../primitives/index';
import { Sidebar } from './Sidebar';
import Button, { ActionButtons } from '../Button';
import Pane from '../Pane';

# Sidebar

## Import

`import { Sidebar } from 'fannypack'`

## Basic Usage

The `<Sidebar>` component uses a fixed `<Overlay>` internally, but it is rendered inside a portal with it's focus trapped (keyboard navigation) for accessibility and also includes a backdrop.

It is recommended to use the sidebar helper components (`<Sidebar.Show>`, `<Sidebar.Hide>`) when building sidebars to leverage accessibility features they provide like so:

<Playground>
  <Sidebar.Container>
    {sidebar => (
      <Box>
        <Button use={Sidebar.Show} {...sidebar}>
          Click me
        </Button>
        <Sidebar {...sidebar}>
          Hello world!
        </Sidebar>
      </Box>
    )}
  </Sidebar.Container>
</Playground>

## Align

You can align the sidebar with the `align` prop which will align the sidebar to the one side of the screen. This can be either set to `"left"` or `"right"`.

<Playground>
  <Sidebar.Container>
    {sidebar => (
      <Box>
        <Button use={Sidebar.Show} {...sidebar}>
          Click me
        </Button>
        <Sidebar align="right" {...sidebar}>
          Hello world!
        </Sidebar>
      </Box>
    )}
  </Sidebar.Container>
</Playground>

## With initial focus set

<Playground>
  <Sidebar.Container>
    {sidebar => (
      <Box>
        <Button use={Sidebar.Show} {...sidebar}>
          Click me
        </Button>
        <Sidebar align="right" {...sidebar}>
          {({ initialFocusRef }) => (
            <React.Fragment>
              Hello world!
              <Button elementRef={initialFocusRef} use={Sidebar.Hide} {...sidebar}>
                Close
              </Button>
            </React.Fragment>
          )}
        </Sidebar>
      </Box>
    )}
  </Sidebar.Container>
</Playground>

## With close button hidden

You can hide the close button with the `hideCloseButton` prop.

<Playground>
  <Sidebar.Container>
    {sidebar => (
      <Box>
        <Button use={Sidebar.Show} {...sidebar}>
          Click me
        </Button>
        <Sidebar align="right" hideCloseButton {...sidebar}>
          Hello world!
          <Button use={Sidebar.Hide} {...sidebar}>
            Close
          </Button>
        </Sidebar>
      </Box>
    )}
  </Sidebar.Container>
</Playground>

## With no animation

You can disable the default animations with the `slide` and `backdropFade` props.

<Playground>
  <Sidebar.Container>
    {sidebar => (
      <Box>
        <Button use={Sidebar.Show} {...sidebar}>
          Click me
        </Button>
        <Sidebar slide={false} backdropFade={false} {...sidebar}>
          Hello world!
        </Sidebar>
      </Box>
    )}
  </Sidebar.Container>
</Playground>

## `<Sidebar>` Props

<PropsTable component={Sidebar} />

## Theming

### Schema

```jsx
{
  Sidebar: {
    base: css | Object,
    Hide: {
      base: css | Object
    },
    Show: {
      base: css | Object
    },
    Close: {
      base: css | Object
    }
  }
}
```
