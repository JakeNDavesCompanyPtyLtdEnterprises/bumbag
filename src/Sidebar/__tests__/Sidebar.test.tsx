import * as React from 'react';
import render from '../../_utils/tests/render';
import { Box } from '../../primitives';
import Sidebar from '../Sidebar';

jest.mock('reakit/Portal', () => 'portal');

it('renders correctly for a default Sidebar', () => {
  const { container } = render(
    <Sidebar.Container defaultVisible>
      {sidebar => <Sidebar {...sidebar}>This is the content</Sidebar>}
    </Sidebar.Container>,
    { theme: { Toast: { disabled: true } } }
  );
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for a Sidebar with align prop set', () => {
  const { container } = render(
    <Sidebar.Container defaultVisible>
      {sidebar => (
        <Sidebar align="right" {...sidebar}>
          This is the content
        </Sidebar>
      )}
    </Sidebar.Container>,
    { theme: { Toast: { disabled: true } } }
  );
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for a Sidebar with hideCloseButton prop set', () => {
  const { container } = render(
    <Sidebar.Container defaultVisible>
      {sidebar => (
        <Sidebar hideCloseButton {...sidebar}>
          This is the content
        </Sidebar>
      )}
    </Sidebar.Container>,
    { theme: { Toast: { disabled: true } } }
  );
  expect(container.firstChild).toMatchSnapshot();
});
