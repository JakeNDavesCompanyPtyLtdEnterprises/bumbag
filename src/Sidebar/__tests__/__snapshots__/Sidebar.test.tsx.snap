// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for a Sidebar with align prop set 1`] = `
.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c3:focus:not(:focus-visible) {
  outline: none;
}

.c10 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: white;
  color: inherit;
}

.c10:focus:not(:focus-visible) {
  outline: none;
}

.c12 {
  fill: inherit;
  position: relative;
  font-size: 1em;
  height: 1em;
  width: 1em;
}

.c9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: white;
  border-radius: 4px;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  min-height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
}

.c9:focus {
  outline: unset;
  z-index: 2;
  box-shadow: 0px 0px 0px 2px;
}

.c9[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c9 {
  border: 0;
  background: unset;
  color: white;
  fill: white;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c9:hover {
  color: #808080;
  fill: #808080;
}

.c2 {
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
  -webkit-transform-origin: calc(50% + 0px) calc(50% + 0px);
  -ms-transform-origin: calc(50% + 0px) calc(50% + 0px);
  transform-origin: calc(50% + 0px) calc(50% + 0px);
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c2[aria-hidden="true"] {
  pointer-events: none;
  opacity: 0;
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
  visibility: hidden;
  will-change: transform,opacity;
}

.c7 {
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
  -webkit-transform-origin: calc(50% + 0px) calc(50% + 0px);
  -ms-transform-origin: calc(50% + 0px) calc(50% + 0px);
  transform-origin: calc(50% + 0px) calc(50% + 0px);
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c7[aria-hidden="true"] {
  pointer-events: none;
  -webkit-transform: translate3d(calc(100% + 0px),0px,0px);
  -ms-transform: translate3d(calc(100% + 0px),0px,0px);
  transform: translate3d(calc(100% + 0px),0px,0px);
  visibility: hidden;
  will-change: transform,opacity;
}

.c1 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 998;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
  -moz-tap-highlight-color: rgba(0,0,0,0);
}

.c6 {
  position: fixed;
  z-index: 19900410;
  left: 50%;
  top: 50%;
}

.c5 {
  top: 0;
  height: 100vh;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
  overflow: auto;
  left: auto;
  right: 0;
}

.c11 {
  border: 0px;
  -webkit-clip: rect(0px,0px,0px,0px);
  clip: rect(0px,0px,0px,0px);
  height: 1px;
  width: 1px;
  margin: -1px;
  padding: 0px;
  overflow: hidden;
  position: absolute;
}

.c0 {
  background-color: black;
  opacity: 0.6;
}

.c8 {
  position: fixed;
  min-height: 24px;
  padding: 0 0.25rem;
  top: 0.5rem;
  right: calc(250px + 0.5rem);
}

.c8:hover {
  color: #f2f2f2;
  fill: #f2f2f2;
}

.c4 {
  background-color: white;
  box-shadow: -3px 0px 12px 0px rgba(0,0,0,0.1);
  width: 250px;
}

<div
  class=""
>
  <div>
    <button
      aria-hidden="false"
      class="c0 c1 c2 c3 "
      role="button"
      tabindex="-1"
    />
    <div
      class="c3"
    >
      <div
        aria-hidden="false"
        aria-modal="true"
        class="c4 c5 c6 c7 c3"
        role="dialog"
        style="width: 250px;"
        width="250px"
      >
        <button
          class="c8 c9 c10 "
          kind="link"
          type="button"
        >
          <div
            class="c11 c3"
          >
            Close
          </div>
          <svg
            aria-hidden="true"
            class="c12 c3"
            role="img"
            size="1"
            style="size: 1px;"
            viewBox="0 0 352 512"
          >
            <path
              d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </button>
        This is the content
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly for a Sidebar with hideCloseButton prop set 1`] = `
.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c3:focus:not(:focus-visible) {
  outline: none;
}

.c2 {
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
  -webkit-transform-origin: calc(50% + 0px) calc(50% + 0px);
  -ms-transform-origin: calc(50% + 0px) calc(50% + 0px);
  transform-origin: calc(50% + 0px) calc(50% + 0px);
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c2[aria-hidden="true"] {
  pointer-events: none;
  opacity: 0;
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
  visibility: hidden;
  will-change: transform,opacity;
}

.c7 {
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
  -webkit-transform-origin: calc(50% + 0px) calc(50% + 0px);
  -ms-transform-origin: calc(50% + 0px) calc(50% + 0px);
  transform-origin: calc(50% + 0px) calc(50% + 0px);
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c7[aria-hidden="true"] {
  pointer-events: none;
  -webkit-transform: translate3d(calc(-100% + 0px),0px,0px);
  -ms-transform: translate3d(calc(-100% + 0px),0px,0px);
  transform: translate3d(calc(-100% + 0px),0px,0px);
  visibility: hidden;
  will-change: transform,opacity;
}

.c1 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 998;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
  -moz-tap-highlight-color: rgba(0,0,0,0);
}

.c6 {
  position: fixed;
  z-index: 19900410;
  left: 50%;
  top: 50%;
}

.c5 {
  top: 0;
  height: 100vh;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
  overflow: auto;
  left: 0;
  right: auto;
}

.c0 {
  background-color: black;
  opacity: 0.6;
}

.c4 {
  background-color: white;
  box-shadow: 3px 0px 12px 0px rgba(0,0,0,0.1);
  width: 250px;
}

<div
  class=""
>
  <div>
    <button
      aria-hidden="false"
      class="c0 c1 c2 c3 "
      role="button"
      tabindex="-1"
    />
    <div
      class="c3"
    >
      <div
        aria-hidden="false"
        aria-modal="true"
        class="c4 c5 c6 c7 c3"
        role="dialog"
        style="width: 250px;"
        width="250px"
      >
        This is the content
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly for a default Sidebar 1`] = `
.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c3:focus:not(:focus-visible) {
  outline: none;
}

.c10 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: white;
  color: inherit;
}

.c10:focus:not(:focus-visible) {
  outline: none;
}

.c12 {
  fill: inherit;
  position: relative;
  font-size: 1em;
  height: 1em;
  width: 1em;
}

.c9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: white;
  border-radius: 4px;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  min-height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
}

.c9:focus {
  outline: unset;
  z-index: 2;
  box-shadow: 0px 0px 0px 2px;
}

.c9[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c9 {
  border: 0;
  background: unset;
  color: white;
  fill: white;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c9:hover {
  color: #808080;
  fill: #808080;
}

.c2 {
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
  -webkit-transform-origin: calc(50% + 0px) calc(50% + 0px);
  -ms-transform-origin: calc(50% + 0px) calc(50% + 0px);
  transform-origin: calc(50% + 0px) calc(50% + 0px);
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c2[aria-hidden="true"] {
  pointer-events: none;
  opacity: 0;
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
  visibility: hidden;
  will-change: transform,opacity;
}

.c7 {
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
  -webkit-transform-origin: calc(50% + 0px) calc(50% + 0px);
  -ms-transform-origin: calc(50% + 0px) calc(50% + 0px);
  transform-origin: calc(50% + 0px) calc(50% + 0px);
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c7[aria-hidden="true"] {
  pointer-events: none;
  -webkit-transform: translate3d(calc(-100% + 0px),0px,0px);
  -ms-transform: translate3d(calc(-100% + 0px),0px,0px);
  transform: translate3d(calc(-100% + 0px),0px,0px);
  visibility: hidden;
  will-change: transform,opacity;
}

.c1 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 998;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
  -moz-tap-highlight-color: rgba(0,0,0,0);
}

.c6 {
  position: fixed;
  z-index: 19900410;
  left: 50%;
  top: 50%;
}

.c5 {
  top: 0;
  height: 100vh;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
  overflow: auto;
  left: 0;
  right: auto;
}

.c11 {
  border: 0px;
  -webkit-clip: rect(0px,0px,0px,0px);
  clip: rect(0px,0px,0px,0px);
  height: 1px;
  width: 1px;
  margin: -1px;
  padding: 0px;
  overflow: hidden;
  position: absolute;
}

.c0 {
  background-color: black;
  opacity: 0.6;
}

.c8 {
  position: fixed;
  min-height: 24px;
  padding: 0 0.25rem;
  top: 0.5rem;
  left: calc(250px + 0.5rem);
}

.c8:hover {
  color: #f2f2f2;
  fill: #f2f2f2;
}

.c4 {
  background-color: white;
  box-shadow: 3px 0px 12px 0px rgba(0,0,0,0.1);
  width: 250px;
}

<div
  class=""
>
  <div>
    <button
      aria-hidden="false"
      class="c0 c1 c2 c3 "
      role="button"
      tabindex="-1"
    />
    <div
      class="c3"
    >
      <div
        aria-hidden="false"
        aria-modal="true"
        class="c4 c5 c6 c7 c3"
        role="dialog"
        style="width: 250px;"
        width="250px"
      >
        <button
          class="c8 c9 c10 "
          kind="link"
          type="button"
        >
          <div
            class="c11 c3"
          >
            Close
          </div>
          <svg
            aria-hidden="true"
            class="c12 c3"
            role="img"
            size="1"
            style="size: 1px;"
            viewBox="0 0 352 512"
          >
            <path
              d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </button>
        This is the content
      </div>
    </div>
  </div>
</div>
`;
